clients/factory.go

```diff
@@ -254,6 +254,19 @@ func (f factory) CreateClassicClient() (*rest.Client, error) {
 	return f.createRestClient(f.classicURL, auth.NewAPITokenClient(context.TODO(), f.accessToken))
 }
 
+// CreateClassicClientWithContext creates a REST client configured for accessing classic APIs with a given context.
```

Comment:
For the implementation of `CreateClassicClient(...)`, what about just calling this method with a `context.TODO()` to prevent having to duplicate the checks?